### PR TITLE
Week13

### PR DESCRIPTION
**Pull request by ruhankhandakar**

### CODE REVIEW

File: Week-13/weather/services/weather.js [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+  const {
+    cityName: queryCityName,
+    cityCode,
+    pageNumber = 1,
+    days,
+    date,
+    hour,
+  } = query;
+
+  if (days && (+days > 10 || +days < 0)) {
```
Comment (ShashankSuresh) [inline Week-13/weather/services/weather.js:39]: Can this be handled in the front end itself? We can restrict making an API call. Kindly share your thoughts.

File: Week-13/weather/middlewares/asyncHandler.js [hunk 2]
```diff
@@ -0,0 +1,4 @@
+const asyncHandler = (fn) => (req, res, next) =>
```
Comment (ShashankSuresh) [inline Week-13/weather/middlewares/asyncHandler.js:1]: Will this handle multiple parallel requests?

File: Week-13/weather/utils/errorResponse.js [hunk 3]
```diff
@@ -0,0 +1,8 @@
+class ErrorResponse extends Error {
+  constructor(message, statusCode) {
```
Comment (ShashankSuresh) [inline Week-13/weather/utils/errorResponse.js:2]: Class based component? Can we have a functional component instead?

File: Week-13/weather/services/weather.js (hunk 1 above)
Comment (ruhankhandakar) [reply Week-13/weather/services/weather.js:39]: Yes, but we don't have UI right now, I think this is backend assignment only

File: Week-13/weather/middlewares/asyncHandler.js (hunk 2 above)
Comment (ruhankhandakar) [reply Week-13/weather/middlewares/asyncHandler.js:1]: this is kind of try-catch wrapper.

File: Week-13/weather/utils/errorResponse.js (hunk 3 above)
Comment (ruhankhandakar) [reply Week-13/weather/utils/errorResponse.js:2]: I think I can't, as Error is also class, and I just created custom Error by extending it.

**Review by ShashankSuresh — APPROVED** [review]
Good job. Approved :)

**Comment by ruhankhandakar** [comment]
> Good job. Approved :)

Thank you 😊

**Pull request merged**